Pubby class remake

spockye opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
Replaces the Current Pubby class carrier with a new version. 
![image](https://user-images.githubusercontent.com/79304582/171997446-0bba99b0-7d4e-47ca-8ef2-05e9243c2a22.png)

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
the current Pubby class carrier is outdated, and barely works. this version brings it up to speed, and makes it more tolerable

****
## Changelog
:cl:
add: Remade the pubby class light carrier
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

spockye committed:
Pubby Class Modernizing

remakes, and fixed the Pubby class Carrier.

spockye committed:
Update whiteship_pubby.dmm

spockye committed:
Update whiteship_pubby.dmm

MusselMain commented:
Nice! I'm all for making shiptests feel more lived in and realistic (at least in a 'this is a long range spaceship that probably needs beds or bunkbeds for their crew')

Apogee-dev commented:
Looks like you still need fire alarms and firelocks. Also, you should probably change the miner outfit to the Asteroid Miner one, since the default for miner is Nanotrasen-themed. Same goes for any miner hardsuits you might have on the map, use the indie one instead. I really need to update the miner outfits to just make indie the default… You also might want to add two Deckhand slots or so. The Pubby’s a small ship, but a default limit of four players is small even for that.

Other than that the only other changes I’d suggest are adding a prefix and namelists in the config.

spockye committed:
fixes missing pipe, and adds firelocks

im gon kil u wit a rok

spockye committed:
adds fire alarms

mmmh, fish jenga

spockye committed:
fixes double and unconnected pipes

how did i miss that

spockye committed:
fixed a wrong layered vent

bonk

spockye committed:
fixes and adds

namelist / prefix / miner -- asteroid miner / added a engineer

spockye committed:
Update pubby.json

spockye committed:
fixes prefix/adds station engineer

spockye commented:
alright, it should be fine and Dandy now. 
---
the crew will now be:
1x captain
1x asteroid miner
1x engineer
2x pilots
____
fire alarms and firelocks have been added, the hardsuits and shaft miner are now independent, a independent prefix has been added, and a namelist

Apogee-dev commented on _maps/configs/pubby.json:
```diff
@@ -1,15 +1,22 @@
 	"map_name": "Pubby-class Light Carrier",
+	"prefix": "ISV",
+	"namelists": ["GENERAL", "BRITISH_NAVY", "MYTHOLOGICAL", "WEAPONS"],
 	"map_short_name": "Punny-class",
```
"map_short_name": "Pubby-class",

while amusing, the shortname actually shows up in the ship purchase menu and a few other places and should actually match the ship's name.

spockye replied:
oh,pft i didnt set it to be like that. it's always been that way

spockye committed:
Update pubby.json

spockye committed:
Missing/wrong locker names

i hate you mapdifbot2

from now on i declare you as my archnemisis

Latentish commented:
I like this pubby more than the old one.

PrefabQuasar commented:
agreed, this looks like a really good remap

spockye committed:
Updates the engine room

some more soul, and ease of use

also adds stock parts and a cargo lathe

removes prebuilt sci machines, and instead places them as boards

More junk and decalling

spockye committed:
Fixes engine wall bending

wall turns the wrong way. solution? Remove

spockye committed:
changes pump location

current ones looked weird.

sorry for the commit spam

spockye committed:
fixes pipes

pipes fuckywucky

Derpguy3 commented:
Space pods need an air canister to use for cabin internals, make sure there is one for the other frame

spockye committed:
adds a air canister

For the second pod, as a fix to Deprguy3's comment

Apogee-dev commented:
Can you add shutters for the windows and engines? Make sure they start out closed so players can't miss the fact that they're there.

spockye committed:
adds shutters

adds shutters to the engines, and a button linked to them. both engines share one button

spockye commented:
oh god why is it all failing

Apogee-dev commented:
> oh god why is it all failing

the .dmm isn't parsing, something fucked up in the file itself. It could be that you accidentally left a string open or added duplicate quotes somewhere when adding the shutters?

spockye committed:
hopefully fixes shit

cmon ship, be Tubular, and fix yerself

spockye committed:
moves the ship file

hopefully fixes the fucky wucky

spockye committed:
final fix

work or i'll track you down.

spockye committed:
Merge branch 'master' into Pubby-class-remake

spockye committed:
Merge branch 'master' into Pubby-class-remake

spockye committed:
Changes the ships faction to NT

spockye committed:
fixes some clothing, and pod

humma humma, darn you balancing!

Apogee-dev commented on _maps/configs/pubby.json:
```diff
@@ -1,15 +1,26 @@
+			"slots": 1
+		},
+		"Pilot": {
+			"outfit":"/datum/outfit/job/cargo_tech/pilot",
```
You'll want to make a new outfit datum here, this one appears to have been made for the Tartarus or some other ship and isn't really NT themed. My recommendation would be to make it security/nt/pilot, or to use the normal security/nt outfit while keeping the Pilot job title.

Some of the locker contents are also mismatched with the new Nanotrasen theme. Use under/rank/security/officer/nt for the officer jumpsuits in the pilot locker (bluesec shirts are the indie ones) and ditch the dusters, they're a strongly indie flavor. For the firearm locker, replace the M1911s and their magazines with Commanders and maybe give the captain an egun if you want his gun to be different from the rest of the crew.

You also still have independent mining hardsuits in the SSUs, you can replace those with normal mining suits.

spockye committed:
Requests from Apogee

did that donzo that, boom!

PrefabQuasar commented:
couple of notes on this:
it would be nice if you expanded the rear airlock to let people drag crates and etc in with them
the picture shown has buttons and fire alarms not properly rotated to fit the walls, if you've already fixed that, ignore this
consider adding light switches

Apogee-dev commented:
You've got a public airlock leading into dorms. Always use the public/glass subtype for these, the basic type is, well. ugly as sin.

spockye committed:
adds a tank holder, and increases the size of the back airlock

spockye committed:
fixed the door

heresy!

spockye committed:
removes some railings from dorms

Derpguy3 commented:
A few things I had noticed while testing out the remake.

- Pod hanger is missing an air alarm.
- Missing a pod circuit and pod core to finish building the second pod.

Latentish commented:
> A few things I had noticed while testing out the remake.
> 
> * Pod hanger is missing an air alarm.
> * Missing a pod circuit and pod core to finish building the second pod.

We shouldn't have issues marging this once you address this.

spockye committed:
adds some more stuffaroonies

adds: pod circuit, pod core, egg box, spacepod chair module, air alarm

removes: Half my sanity, 3 years cut from my lifespan, a bragain with the devil

Latentish reviewed (approved):
Good changes. I especially like the json changes to bring it up to more recent standards of how these files are written. Looking real margable.
